Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java

```diff
@@ -3,23 +3,28 @@
 
 import com.avangers.backendapi.DTOs.RegisterUserDTO;
 import com.avangers.backendapi.services.UserService;
+import com.avangers.backendapi.services.UserServiceResponse;
 import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.CrossOrigin;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
-@CrossOrigin("http://localhost:5173/")
 @RestController
+@CrossOrigin("http://localhost:5173/")
 @RequiredArgsConstructor
 public class UserController {
 
     private final UserService userService;
 
     @PostMapping("/register")
     public ResponseEntity<String> register(@Valid @RequestBody RegisterUserDTO registerUserDTO) {
-        return userService.addUser(registerUserDTO);
+        UserServiceResponse response = userService.addUser(registerUserDTO);
```

Comment:
I would rather throw a custom made error and handle it with a controlleradvice some place else.
https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc